wordpress-mobile/WordPress-iOS
# Making Post Card adapt to bigger font sizes.

This PR fixes an issue on the Post List, where the layout was broken when using big font sizes.

![post-list-accessibility](https://user-images.githubusercontent.com/9772967/51112971-d12d8a00-1800-11e9-8fe9-1814e30102b5.png)

**Notes:**
My first strategy was to replace many of the manually added constraints with StackView, since it's a lot easier to manage self sizing views with them.

Stack Views also makes it trivial to adapt the UI Layout when showing/hiding views, so all the code managing constraints became not necessary.

This is also true for the featured image view, just by assigning `isHided = true`, the layout will adapt as the view was never there, making the `PostCardImageCell.xib` unnecessary. (As noted by @pinarol )

A last detail, using the power of stack views was to layout the time label with the metadata labels vertically when bigger fonts are on use. This gives more horizontal space for the labels to show it's content properly. (cc @iamthomasbishop )

Hopefully with these changes it will be easier in the future to add/remove elements to the post cards, since they are now based mostly on StackView and there is just one `xib` to modify. It will also be more difficult to make them not adaptable to dynamic font sizes.

**To test:**

- Check that the post list displays correctly with normal font size.
  - Be sure to test posts with and without images.
  - Also test posts on sites that will and won't display the header (personal site vs sites with multiple users)
- Increase the font size to one of the accessibility font sizes.
  - iOS Settings -> General -> Accessibility -> Larger Text -> Turn the switch ON -> Increase the font size.
- Make the same test than before.
